cf/field.py

```diff
@@ -13040,6 +13044,174 @@ def domain_mask(self, **kwargs):
 
         return mask
 
+    @_inplace_enabled(default=False)
+    @_manage_log_level_via_verbosity
+    def compute_vertical_coordinates(self, default_to_zero=True,
+                                     strict=True, inplace=False,
+                                     verbose=None):
+        '''Compute non-parametric vertical coordinates.
+
+    When vertical coordinates are a function of horizontal location as
+    well as parameters which depend on vertical location, they cannot
+    be stored in a vertical dimension coordinate construct. In such
+    cases a parametric vertical dimension coordinate construct is
+    stored and a coordinate reference construct contains the formula
+    for computing the required non-parametric vertical
+    coordinates.
+
+    For example, multi-dimensional non-parametric parametric ocean
+    altitude coordinates can be computed from one-dimensional
+    parametric ocean sigma coordinates. See the "Parametric Vertical
+    Coordinate" sections of the CF conventions for more details.
```

Comment:
Hi Sadie, I'm in two minds about this explicit links. On one hand, explicit links are useful, on the other, they would resolve to the _latest_ conventions; the docs and code being may be for an earlier version, and the conventions may have changed since that time. Also, the version-less links are for the latest draft, rather than the latest official. 

It's just occured to me that we could use docstring rewriting to pin the version down, and include URLs that are guaranteed to be appropriate:

```
https://cfconventions.org/Data/cf-conventions/cf-conventions-{{VN}}/cf-conventions.html
```

This would be automatically up to date, because we would define {{VN}} as follows
```python
# cf/docstring/docstring.py (or better - put it in cfdm's doctring.py)
from .. import __version__
_docstring_substitution_definitions = {
  '{{VN}}': __version__,
}
``` 
What do you think?

Reply:
Aha, sounds like a great solution! If it's simple enough to implement then please go for it (in a separate PR if you prefer). If not, I am happy for you to decide what's best. I did think there would be some reason you hadn't put the links up and linking to the out-of-date conventions would probably be less desirable than having no links so the user has to look it up if they want to know more.

Reply:
I'll do it this PR, I think. I'll also class-ify the function in formula_terms.py so they can benefit, and also do `{{default_to_zero: `bool`, optional}}`

Reply:
Resolved in https://github.com/NCAS-CMS/cf-python/pull/158/commits/c0ea6a3db284873c0ce9a830b268b7748d87e8ac